WordPress/gutenberg
# List View: Allow Escape key to deselect blocks if blocks are selected

## What?


Fixes #48462 

## Related issues and discussions:

* https://github.com/WordPress/gutenberg/issues/36535
* https://github.com/WordPress/gutenberg/issues/47172

This is an attempt to solve the issue of users being unable to deselect blocks from within the list view. The idea proposed is to allow the Escape key to deselect all blocks if there is a block selection, and skip closing the list view. If there are no blocks selected, then the Escape key will close the list view.

## Why?


As reported in #48462, it would be good for users to be able to deselect blocks from within the list view via keyboard, and it is not currently possible to do so. Because folks can use the list view keyboard shortcut to open and close it, the idea in this PR is that it would be okay for Escape to be prioritised for deselecting blocks rather than closing the list view.

For reference, the keyboard shortcut for closing the list view is alt+shift+o on Windows, and control+option+o on Mac.

## How?


* In the list view if there is a block selection when the user presses the Escape key, then clear the block selection instead of closing the list view. Otherwise, proceed to closing the list view.

## Testing Instructions





1. In a post with multiple blocks, try shift selecting a couple of them block from the list view.
2. Press Escape and the block should be deselected.
3. Press Escape again and the list view should be closed.
4. Open the list view again, and tab to the items within the list view without selecting any of the blocks. Press Escape and the list view should close.

### Testing Instructions for Keyboard


1. In a post with multiple blocks, open the list view via the keyboard shortcut (alt+shift+o on Windows, and control+option+o on Mac).
2. Use shift + up/down keys to select multiple blocks.
3. Press Escape and the blocks should be deselected.
4. Press Escape again and the list view should be closed.

## Screenshots or screencast 

In the below screengrab, the user shift-clicks multiple blocks within the list view, then presses Escape to clear the selection. When there are no blocks selected and the user presses Escape, the list view is closed.

https://user-images.githubusercontent.com/14988353/222635879-f751e946-9cde-403e-9383-ea45126f09ec.mp4
